admin_app/src/app/user-management/page.tsx

```diff
@@ -0,0 +1,199 @@
+    }
+  }, [recoveryCodes]);
+  const handleRegisterModalContinue = (newRecoveryCodes: string[]) => {
+    setRecoveryCodes(newRecoveryCodes);
+    setLoading(true);
+    setShowCreateModal(false);
+  };
+  const handleEditModalContinue = (newRecoveryCodes: string[]) => {
+    setLoading(true);
+    setShowEditModal(false);
+  };
+
+  const handleResetPassword = (user: UserBody) => {
+    setCurrentUser(user);
+    setShowUserResetModal(true);
+  };
+
+  const handleEditUser = (user: UserBody) => {
+    setCurrentUser(user);
+    setShowEditModal(true);
+  };
+
+  if (role !== "admin") {
+    return (
+      <Box display="flex" justifyContent="center" alignItems="center" height="100vh">
+        <Typography variant="h4" color="error">
+          [403] Access Denied
+        </Typography>
+      </Box>
+    );
+  }
+
+  return (
+    <Grid container sx={{ padding: 3 }}>
+      <Grid
+        item
+        xs={12}
+        sx={{
+          display: "flex",
+          justifyContent: "space-between",
+          alignItems: "center",
+          paddingBottom: 2,
+        }}
+      >
+        <Box>
+          <Typography variant="h4">Manage User</Typography>
+          <Typography variant="body1" align="left" color={appColors.darkGrey}>
+            Add and edit user passwords.
+          </Typography>
+        </Box>
+        <Button
+          variant="contained"
+          color="primary"
+          onClick={() => {
+            setShowCreateModal(true);
+          }}
+        >
+          Add User
+        </Button>
+      </Grid>
+      <Grid item xs={12}>
+        {loading ? (
+          <Box display="flex" justifyContent="center" alignItems="center" height="100%">
+            <CircularProgress size={24} />
+          </Box>
+        ) : (
+          <Paper
+            elevation={0}
+            sx={{
+              marginLeft: 10,
+              marginRight: 10,
+              paddingTop: 1,
+              paddingBottom: 10,
+              paddingInline: 3,
+              minHeight: "65vh",
+              overflowY: "auto",
+              border: 0.5,
+              borderColor: "lightgrey",
+            }}
+          >
+            <List>
+              {users.map((user, index) => (
+                <Layout.FlexBox key={user.user_id}>
+                  <UserCardNew
+                    index={index}
+                    username={user.username}
+                    isAdmin={user.is_admin}
+                    isLastItem={index === users.length - 1}
+                    hoveredIndex={hoveredIndex}
+                    setHoveredIndex={setHoveredIndex}
+                    onResetPassword={() => handleResetPassword(user)}
+                    onEditUser={() => handleEditUser(user)}
+                  />
+                  <EditUserModal
+                    open={showEditModal}
+                    onClose={() => {
+                      setShowEditModal(false);
+                    }}
+                    user={currentUser!}
```

Comment:
If I go to edit or add user and then hit cancel, it gives me a 403 and I can't access `User management` again

Reply:
Fixed!